### DISCUSSION TITLE
KaTeX Autonumbering and User Defined CSS Counter

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6607] [General]
I have encountered an issue in using automatic numbering using CSS counter when I use the KaTex official plugin.
KaTeX plugin in `$:/plugins/tiddlywiki/katex/styles` uses `.tc-tiddler-frame { counter-reset: katexEqnNo; }` to allow correct numbering of equations.

If you have another plugin like Refnote which it also uses CSS numbering with `.tc-tiddler-frame { counter-reset: fnote-count;}` then as two or more counter reset under `.tc-tiddler-frame` does not work then one of plugin shows wrong numbering for example see https://github.com/kookma/TW-Refnotes/issues/23#issuecomment-1088334577

My investigation shows we need to have all counter reset in one declaration statement see https://stackoverflow.com/questions/29126881/more-counters-in-css

So, adding a new stylesheet with

```
.tc-tiddler-frame { counter-reset: fnote-count katexEqnNo;}
```

Solve the issue.

Question: What solution do you propose for Tiddlywiki? One solution is given above, but it is not intuitive.

**Comment by kookma** [comment]
Further investigation

(1)

I noticed when both KaTeX and Refnotes plugins are installed, reordering the stylesheets works and solves the issue. e.g. putting the Refnotes styles 

`$:/plugins/kookma/refnotes/styles/footnote-counter`

after KaTeX style

`$:/plugins/tiddlywiki/katex/styles`

So, is there anything wrong in `$:/plugins/tiddlywiki/katex/styles`?

(2)
I noticed having both plugins in a Tiddlywiki only breaks the numbering for Refnotes (footnote) if an empty line is used between two footnote for example

```
a<<fnote "One">>
this is a line
b<<fnote "Two">>
c<<fnote "Three">>d<<fnote "Four">>
```

renders as (a<sup>1</sup> this is a line b<sup>2</sup> c<sup>3</sup>), which works correctly, but 

```
a<<fnote "One">>

b<<fnote "Two">>
c<<fnote "Three">>d<<fnote "Four">>
```

does not work and renders as

a<sup>1</sup> 

b<sup>1</sup> c<sup>2</sup>

Number is reset when a new line is started.

Disabling the KaTeX makes Refnotes works fine.

**Reply by linonetwo** [reply]
why counters will have a collision? It seems to have a counter id.

**Reply by kookma** [reply]
Not sure, but see https://stackoverflow.com/questions/29126881/more-counters-in-css
by the way the solution given above solve the problem! e.g `.tc-tiddler-frame { counter-reset: fnote-count katexEqnNo;}`
